### PR TITLE
Update go 1.14

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
* Update docker images to use Go 1.14
* Migrate from dep to go modules

### CODE REVIEW

**Review by sleshchenko — APPROVED** [review]
I tested nothing but changes LGTM

**Comment by sleshchenko** [comment]
:warning: please check why 

> ci/circleci: build — Your tests failed on CircleCI

**Comment by amisevsk** [comment]
> warning please check why
>
>    ci/circleci: build — Your tests failed on CircleCI

You're quick :D -- I've been trying to fix circleCI for the last 15 minutes, the failures make no sense (setting workingDirectory -> no modules matched by `./...`)

File: build/artifacts/rhel.Dockerfile [hunk 1]
```diff
@@ -10,14 +10,14 @@
...
-WORKDIR /go/src/github.com/eclipse/che-plugin-broker/brokers/artifacts/cmd/
-COPY . /go/src/github.com/eclipse/che-plugin-broker/
+WORKDIR /build/che-plugin-broker/brokers/artifacts/cmd/
+COPY . /build/che-plugin-broker/
 RUN adduser appuser && \
-    CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-w -s' -installsuffix cgo -o artifacts-broker main.go
+    CGO_ENABLED=0 GOOS=linux go build -mod vendor -a -ldflags '-w -s' -installsuffix cgo -o artifacts-broker main.go
 
 # https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
 FROM registry.access.redhat.com/ubi8-minimal:8.2-267
```
Comment (nickboldt) [inline build/artifacts/rhel.Dockerfile:23]: latest tag at https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal is...
`8.3-230`

File: build/dev/Dockerfile [hunk 2]
```diff
@@ -10,13 +10,10 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-FROM quay.io/eclipse/che-golang-1.12:nightly
+FROM quay.io/eclipse/che-golang-1.14:nightly
```
Comment (nickboldt) [inline build/dev/Dockerfile:13]: woo, skipping 1.13 and going directly to 1.14! #consistentcy-win!

File: build/metadata/rhel.Dockerfile [hunk 3]
```diff
@@ -10,14 +10,14 @@
...
-WORKDIR /go/src/github.com/eclipse/che-plugin-broker/brokers/metadata/cmd/
-COPY . /go/src/github.com/eclipse/che-plugin-broker/
+WORKDIR /build/che-plugin-broker/brokers/metadata/cmd/
+COPY . /build/che-plugin-broker/
 RUN adduser appuser && \
-    CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-w -s' -installsuffix cgo -o metadata-broker main.go
+    CGO_ENABLED=0 GOOS=linux go build -mod vendor -a -ldflags '-w -s' -installsuffix cgo -o metadata-broker main.go
 
 # https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
 FROM registry.access.redhat.com/ubi8-minimal:8.2-267
```
Comment (nickboldt) [inline build/metadata/rhel.Dockerfile:23]: latest tag at https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal is...
8.3-230

File: build/dev/Dockerfile (hunk 2 above)
Comment (amisevsk) [reply build/dev/Dockerfile:13]: We never even build che-golang-1.13 as far as I know :smile:

File: build/artifacts/rhel.Dockerfile (hunk 1 above)
Comment (amisevsk) [reply build/artifacts/rhel.Dockerfile]: Thanks! Updated.

File: build/metadata/rhel.Dockerfile (hunk 3 above)
Comment (amisevsk) [reply build/metadata/rhel.Dockerfile]: Thanks! Updated.

File: build/dev/Dockerfile (hunk 2 above)
Comment (amisevsk) [reply build/dev/Dockerfile:13]: In this house we only respect even-numbered Go releases: [1.10](https://quay.io/repository/eclipse/che-golang-1.10), [1.12](https://quay.io/repository/eclipse/che-golang-1.12), [1.14](https://quay.io/repository/eclipse/che-golang-1.14)

**Review by nickboldt — APPROVED** [review]
LGTM, ready to push it and see if it works as expected in downstream :)

**Comment by nickboldt** [comment]
cherry picked to https://github.com/eclipse/che-plugin-broker/commits/v3.4.x too

**Pull request merged**